spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/sns/SnsAutoConfiguration.java

```diff
@@ -70,6 +71,12 @@ public SnsClient snsClient(SnsProperties properties, AwsClientBuilderConfigurer
 				.build();
 	}
 
+	@Bean
+	@ConditionalOnProperty(name = SnsProperties.PREFIX + ".resolve_by_list", havingValue = "true")
```

Comment:
Let's take it out from the auto-configuration. If users want to use it, they can declare the bean themselves.